### PR TITLE
V850: Improve RH850 support, add RzIL uplifting

### LINKED ISSUE #9: Add support for RH850 architecture
Successor of V850 architecture

[RH850G3MH-UserManual-Software.pdf](https://github.com/radare/radare2/files/2200051/RH850G3MH-UserManual-Software.pdf)

Code could be shared with V850 in the same module (`librz/asm/p/asm_v850.c`/`librz/analysis/p/analysis_v850.c`).

See also:
- https://github.com/NationalSecurityAgency/ghidra/tree/master/Ghidra/Processors/V850

### PR DESCRIPTION
**Pull request by b1llow**
# SQUASH ME

**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

V850 and RH850 manual https://www.renesas.com/us/en/document/mah/v850-familytm-architecture

**Test plan**

- CI is green
- `aaa` and `plf @@F` on all samples make sense

**Closing issues**

Closes https://github.com/rizinorg/ideas/issues/9

### CODE REVIEW

**Comment by XVilka** [comment]
You can also check it on the example V850 file from https://github.com/rizinorg/rizin/issues/2316 or https://github.com/rizinorg/rizin/issues/2317

**Review by XVilka — CHANGES_REQUESTED** [review]
In the disassembler (`librz/asm/p/asm_v850.c`), please also use the new `rz_asm_op_setf_asm()` API to reduce one more allocation per instruction (https://github.com/rizinorg/rizin/issues/4127).

**Comment by b1llow** [comment]
I've noticed that there are a few differences between the various minor versions of RH850, but I can't find documentation for them on the web.

**Comment by XVilka** [comment]
Please rebase

**Comment by b1llow** [comment]
Although there are still some rh850 instructions outstanding, emulateme can be executed successfully！

File: librz/analysis/arch/v850/v850_il.c [hunk 1]
```diff
@@ -0,0 +1,1192 @@
...
+	"r25",
+	"r26",
+	"r27",
+	"r28",
+	"r29",
+	"ep", // EP
+	"lp", // LP
+	/*PC*/
+
+	/*0 */ "EIPC",
```
Comment (XVilka) [inline librz/analysis/arch/v850/v850_il.c:45]: Please add documentation here too

File: librz/asm/arch/v850/v850_disas.h [hunk 2]
```diff
@@ -408,66 +401,168 @@ typedef enum {
...
+	return (i->d >> 17) & 0xf;
 }
 
 static inline ut16 get_disp22(const V850_Inst *i) {
-	return (i->w2 & ~1) | ((i->w1 & 0x3f) << 16);
+	return (V850_word(i, 2) & ~1) | ((V850_word(i, 1) & 0x3f) << 16);
+}
+
+static const ut8 V850_list12_map[] = {
+	/*[0]  = */ 30,
```
Comment (XVilka) [inline librz/asm/arch/v850/v850_disas.h:478]: And here

File: test/db/analysis/v850 [hunk 3]
```diff
@@ -5,72 +5,72 @@ aap
 afl
 EOF
 EXPECT=<<EOF
-0x00100018    1 8            fcn.00100018
```
Comment (XVilka) [inline test/db/analysis/v850:8]: Have checked validity of these changes?

**Review by XVilka — CHANGES_REQUESTED** [review]
LGTM overall

File: test/db/analysis/v850 (hunk 3 above)
Comment (b1llow) [reply test/db/analysis/v850:8]: ~I checked some but not all of them with IDA.~

Comment (b1llow) [reply test/db/analysis/v850:8]: Actually, one of the things I've found is that when `aaa` has been used to do a full analysis, the information on the functions is much more accurate.

Comment (XVilka) [reply test/db/analysis/v850:8]: @imbillow accurate compared to what?

Comment (b1llow) [reply test/db/analysis/v850:8]: > @imbillow accurate compared to what?

Compare the output of `afl`, which contains the addr bbs size real_size for each function

**Review by XVilka — COMMENTED** [review]
The V850/RH850 and RzIL parts are good now; also tested but are blocked until https://github.com/rizinorg/rizin/pull/4179 is sorted out.

**Comment by XVilka** [comment]
@imbillow please rebase

**Review by XVilka — APPROVED** [review]

**Pull request merged**